src/iterators.rs

```diff
@@ -24,6 +26,7 @@ impl Iterator for KeyIter {
     }
 }
 
+#[derive(Clone, Debug, PartialEq, PartialOrd, Eq, Ord, Hash)]
```

Comment:
This class already has a clone() method that does something different. I'd rather not add things like Hash until it's needed, for internal classes like this

Reply:
As long as it is always internal **and** it is never owned or referenced by any public class, then that won't be a problem.

Reply:
I don't expect it will be. If it's moved into the public API in the future, I'm happy to implement those derives

Reply:
OK, so removing `Hash` and the comparison traits is easy.  That said, I think you should rename `clone` to something else (probably `to_key_iter`) because anyone reading your code is going to expect `clone` to be an implementation of `Clone`.  When it it isn't, this will cause headaches.